### ISSUE TITLE
Initial integration with Decidim proposals

### ISSUE THREAD

**Issue by oliverbarnes** [liquidvotingio/decidim-module-liquidvoting#27]
User Scenarios:
- [x] participant creates delegation
- [x] participant deletes delegation
- [x] participant votes (https://github.com/liquidvotingio/decidim-liquidvoting/issues/2)
- [x] participant removes vote (https://github.com/liquidvotingio/decidim-liquidvoting/issues/3)
- [x] participant views voting result with the delegations factored in (https://github.com/liquidvotingio/decidim-liquidvoting/issues/4)

Other tasks:
- [x] encapsulate spot LV api queries in a LV state object that controllers acquire at start of request, and views access 
- [x] choose `proposal.proposal_votes_count` approach: abandon or manage (abandon) 
- [x]  ... and then make code consistent
- [ ] ~~Add tests (#16)~~
- [ ] ~~Document our module approach, how we change proposal voting, what decidim proposal functionality is lost ... in the README~~
- [x] add delegation UI back to `ProposalController#show`, insure everything using the LV state obj
- [x] vote counts and button still not refreshing automatically
- [ ] ~in `Decidim::Liquidvoting::Client` we need to fix the `votes` and `delegations` brute force hacks~ (Moved to https://github.com/liquidvotingio/decidim-module-liquidvoting/issues/32)
- [ ] ~ajax interface updates (#18)~ (I propose we unlink from the current issue to unblock it and PR #25)
- [ ] ~add some support/delegation UI love to help user understand if and when their vote is counted (eg display if delegate has voted or not, maybe link out to a delegation explainer)~ (I propose this be a separate issue)

**Comment by jinjagit** [comment]
WIP [g-doc](https://docs.google.com/document/d/12yl5ujjAl91KhfZL_gMIC06WNsuJ3jCfKxL-eCsySj8/edit?usp=sharing) of various vote-button stylings x various vote_button partial conditional states (for unmodified decidim + development_app)

**Comment by jinjagit** [comment]
Add component: proposals  

Currently, we plan to avoid recording each user vote in the decidim `proposals-proposals-votes` table, since this would require either reproducing our api delegation(s) mechanism on the decidim side, or passing data on all votes created/deleted back to decidim whenever a create/delete vote/delegation action is executed.
This means we would need to grey-out / deactivate / hide the 'Support limit per participant' and 'Minimum supports per user' in the admin view when creating the proposals component (or add these limits to our api and get the relevant values / booleans from our api when needed.  
  
![proposals_user_stuff](https://user-images.githubusercontent.com/3944042/104585551-138e3980-565c-11eb-9b75-a53962ea2176.png)

**Comment by davefrey** [comment]
Maybe this is a decent place to summarize/explain the changes I've pushed?

I've introduced a LV state object that encapsulate LV proposal queries like "how many votes". The intent is that controllers acquire this LV state object for the web request, and then views reference it for what used to be point queries to the LV api.

`Decidim::Liquidvoting::Client` changed a lot: I minimized the public api, and restructured the file to more easily read public / private / graphql query definitions. The public api is now just the mutations (create/delete votes and delegations), and a getter for the current LV state object.

**Comment by oliverbarnes** [comment]
Thanks for the detailed update 👍  I'm reviewing the changes, will comment in the PR and we could have a call at some point to discuss the open questions

**Comment by davefrey** [comment]
Re: abandon-or-manage `proposals.proposal_votes_count`: After pairing, we agreed to take the "abandon" approach.
This means we will override all of the ~20 decidim files that reference the attribute, to instead reference LV state.
Also, we won't bother raising exceptions or otherwise try to flush out uses of the attribute that we missed. "Abandon without prejudice" :-)

**Comment by davefrey** [comment]
The checklist includes issue #18 moving voting and delegating to ajax calls; votes are working, delegation is not, I propose we remove #18 from the current issue and from PR #25, and let us track the delegation part on #18 and a new PR.

**Comment by oliverbarnes** [comment]
Sounds good - let's also punt adding tests (#16) to a separate PR.
